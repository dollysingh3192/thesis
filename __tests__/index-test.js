//TODO: fix test runner

const babel = require('babel-core');
const fs = require('fs');
const path = require('path');
// const plugin = require('../packages/babel-plugin/');

const INPUTS_DIR = path.resolve(__dirname, 'inputs');
const OUTPUTS_DIR = path.resolve(__dirname, 'outputs');

const inputFiles = fetchFiles(INPUTS_DIR);

// it('Compilations match snapshots', () => {
//   inputFiles.forEach(({content}) => {
//     console.log(content);
//     const {code} = babel.transform(content, {plugins: [plugin]});
//     expect(code).toMatchSnapshot();
//   })
// })

inputFiles.forEach(({name}) => {
  const inputFile = path.resolve(INPUTS_DIR, name);
  const outputFile = path.resolve(OUTPUTS_DIR, name);
  const input = require(inputFile);
  const output = require(outputFile);

  it(name + ' is functionally equivalent before and after compilation', () => {
    expect(output(10)).toEqual(input(10));
  })
})

function fetchFiles(dir) {
  return fs
          .readdirSync(dir)
          .map(name => (
            {
              name,
              content: fs.readFileSync(path.resolve(dir, name), 'utf8')
            }
          ))
}