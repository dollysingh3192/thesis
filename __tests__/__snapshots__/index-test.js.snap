// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compilations match snapshots 1`] = `
"// example from http://homepages.inf.ed.ac.uk/wadler/papers/deforest/deforest.ps

module.exports = function (n) {
  let upto_n = [];
  for (let i = 0; i < n; i++) upto_n.push(i + 1);

  upto_n.map(_ => (n => {
    val: n;
  })((n => n * n)(_))); // [a] -> [b]   ( a -> b )
};"
`;
